src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -1,9 +1,144 @@
 import React from 'react';
+import Cropper from 'react-cropper';
+import Dropzone from 'react-dropzone';
 
 export default class PhotoField extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      cropResult: null,
+      done: false,
+      zoomValue: 2
+    };
+  }
+
+  onDone = () => {
+    this.setState({ src: null, done: true });
+  }
+
+  onChange = (e) => {
+    e.preventDefault();
+    let files;
+    if (e.dataTransfer) {
+      files = e.dataTransfer.files;
+    } else if (e.target) {
+      files = e.target.files;
+    }
+    const reader = new FileReader();
+    reader.onload = () => {
+      this.setState({ src: reader.result, done: false, cropResult: null });
+    };
+    reader.readAsDataURL(files[0]);
+  }
+
+  onDrop = (files) => {
+    this.setState({ src: files[0].preview });
+  }
+
+  onZoom = (e) => {
+    const newZoomValue = e.detail.ratio;
+    this.setState({ zoomValue: newZoomValue }, () => {
+      this.refs.slider.value = this.state.zoomValue;
+    });
+  }
+
+  moveUp = (e) => {
+    e.preventDefault();
```

Comment:
If you put `type="button"` on the buttons, these prevent default calls shouldn't be necessary.

Reply:
Thanks @jbalboni, change made.